addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -489,35 +489,35 @@ options.registry.layout_column = options.Class.extend({
      * @param {integer} count - positif to add, negative to remove
      */
     _updateColumnCount: function (count) {
-        if (!count) {
+        var delta = count - this.$target.children().length;
+        if (!delta) {
             return;
         }
 
         this.trigger_up('request_history_undo_record', {$target: this.$target});
 
-        if (count > 0) {
+        this._resizeColumns(count);
+        if (delta > 0) {
             var $lastColumn = this.$target.children().last();
-            for (var i = 0; i < count; i++) {
+            for (var i = 0; i < delta; i++) {
                 this.trigger_up('clone_snippet', {$snippet: $lastColumn});
             }
         } else {
             var self = this;
-            _.each(this.$target.children().slice(count), function (el) {
+            _.each(this.$target.children().slice(delta), function (el) {
                 self.trigger_up('remove_snippet', {$snippet: $(el)});
```

Comment:
Add a TODO comment next to this line to indicate we have to consider the async part of this in master :+1